builtin/providers/terraform/provider.go

```diff
@@ -1,21 +1,151 @@
 package terraform
 
 import (
-	"github.com/hashicorp/terraform/helper/schema"
-	"github.com/hashicorp/terraform/terraform"
+	"fmt"
+
+	"github.com/hashicorp/terraform/config/configschema"
+	"github.com/hashicorp/terraform/providers"
+	"github.com/zclconf/go-cty/cty"
 )
 
-// Provider returns a terraform.ResourceProvider.
-func Provider() terraform.ResourceProvider {
-	return &schema.Provider{
-		ResourcesMap: map[string]*schema.Resource{
-			"terraform_remote_state": schema.DataSourceResourceShim(
-				"terraform_remote_state",
-				dataSourceRemoteState(),
-			),
+// Provider is an implementation of providers.Interface
+type Provider struct {
+	// Provider is the schema for the provider itself.
+	Schema providers.Schema
+
+	// DataSources maps the data source name to that data source's schema.
+	DataSources map[string]providers.Schema
+}
+
+// NewProvider returns a new terraform provider
+func NewProvider() *Provider {
+	return &Provider{}
+}
+
+// GetSchema returns the complete schema for the provider.
+func (p *Provider) GetSchema() providers.GetSchemaResponse {
+	return providers.GetSchemaResponse{
+		Provider: providers.Schema{
+			Version: 1,
```

Comment:
@jbardin is it necessary to set this here? I don't think we have any real mechanism to deal with provider-level schema versioning because provider configuration isn't kept in the state anyway, so my instinct would've been to leave this unset (having it default to `0`) but I want to check I'm not missing something here before I recommend that!

Reply:
Nope, the provider doesn't need a version other than the zero value. 
This is just a artifact from having a single schema type to work with in the protocol.